plugins/auth/irods_auth_plugin-native/libirods_auth_plugin-native.cpp

```diff
@@ -0,0 +1,467 @@
+#include "authentication_plugin_framework.hpp"
+
+#include "base64.h"
+
+int get64RandomBytes( char *buf );
+void setSessionSignatureClientside( char* _sig );
+void _rsSetAuthRequestGetChallenge( const char* _c );
```

Comment:
we should probably add a comment explaining from what header file these functions come from.

Reply:
Plot twist: these functions should actually be `extern`'d because they are declared and defined in `.cpp` files. Does this seem correct?

Reply:
i don't think they need the extern keyword since they are declarations.

i believe extern matters when dealing with variables.

Reply:
Indeed, it's not needed.

On that note, do you still think the comments are needed?

Reply:
i'm kind of torn on this.

these forward declarations mean we now have to maintain them in at least two places. comments can also become stale. if the header is included, we don't have to worry about it.

however, i get why someone would do this. maybe jason wanted to avoid including a large header file or this was just easier. avoiding including of a large file isn't a good reason to do this given how fast computers are.

Reply:
These functions are declared *and* defined in `.cpp` files. They are not declared in header files.

Reply:
Oh no. We need to fix that.

Can handle that in a separate PR.

Reply:
I created a new issue for this: #6152